sim/battle.ts

```diff
@@ -1022,15 +1022,12 @@ export class Battle extends Dex.ModdedDex {
 		}
 		}
 
-		let allRequestsMade = true;
-		for (const request of requests) {
-			if (!request) allRequestsMade = false;
-		}
+		let allRequestsMade = !requests.some(request => !request);
 		for (let i = 0; i < this.sides.length; i++) {
 			const side = this.sides[i];
 			const request = requests[i];
 			if (request) {
-				if (!this.supportCancel || allRequestsMade) request.noCancel = true;
+				if (!this.supportCancel || !allRequestsMade) request.noCancel = true;
```

Comment:
Can we add a regression test for this (or fix any cancel tests we already have that should have caught this)?

Reply:
For the record, this only affected whether or not the cancel button actually appeared client-side. It didn't actually stop users from sending `/choose undo` and cancelling their choice.

Reply:
True. So less critical to worry about, but if we could cover it with a test without too much effort itd still be worth doing.

Reply:
Also, basically zero users know about `/choose undo`, so this is approximately as critical; it just means the test needs to be designed to account for this case.

Reply:
I think the naive way of adding the test would be to add a new method to ``test/assert.js`` which allows for asserting on the latest ``request`` object on the battle log for a side (unless theres an easier way to access the request in a test than searching through the battle log)

Reply:
Yeah I was just saying that's why the tests passed.

Reply:
We should probably rewrite the relevant code to make it easier to extract requests; that sounds less effort than screwing around with `test/assert.js`.